.github/settings.sh

```diff
@@ -32,7 +32,7 @@ export BAZEL_CXXOPTS="-std=c++17"
 export BAZEL_OPTS="-c opt --show_progress_rate_limit=10.0"
 
 # Used to fetch the BAZEL version where needed.
-export BAZEL_VERSION=4.0.0
+export BAZEL_VERSION=5.3.2
```

Comment:
Do we need a switch to latest bazel ?
I think we wanted to make sure to be able to run with slightly older versions, so keeping the lowest version that still works is probably good.

Reply:
Just to reiterate what we've already discussed in person: because of the implicit existence of [bazelisk](https://github.com/bazelbuild/bazelisk) on the [GitHub Actions runner image](https://github.com/actions/runner-images/blob/9f92c57e695e8950765b3596e1add8b31d657df6/images/linux/scripts/installers/bazel.sh#L4), the CI in this project hasn't been using the version specified in the `BAZEL_VERSION` variable for quite some time now. What happened was that Bazelisk would update Bazel to the latest version. One particular run actually started failing and upon investigation the Bazel version was the only difference -- the `bazel version` command invoked in the "old" CI did in fact report `5.3.2`, hence I bumped it here and the build started to pass.

Considering that we no longer have Bazelisk (as we did not install it and the container that we use doesn't have it implicitly installed), bumping the version seems to be the good solution here. If need be, we can always install Bazelisk in the future.